### DISCUSSION TITLE
URL's in shortcodes [tsml_types_list] & [tsml_regions_list] are incorrect when using TSML UI as frontend

### DISCUSSION THREAD

**Discussion by anchovie91471** [code4recovery/12-step-meeting-list#1086] [General]
I just noticed this when using the shortcodes on a fresh WP 6.1.1 install with only TSML installed, using the 2023 theme. (I tried 2021 as well)

The links are fine for Legacy UI, but with TSML UI (react), the parameters look like they should be different.

1. **[tsml_types_list]:** The link for "Big Book" is [https://northmsaa.org/testsite/meetings/?tsml-day=any&tsml-type=B](https://northmsaa.org/testsite/meetings/?tsml-day=any&tsml-type=B) but should be (maybe) [https://northmsaa.org/testsite/meetings/?tsml-type=big-book](https://northmsaa.org/testsite/meetings/?tsml-type=big-book). Looks like TSML UI uses a slug instead of the type key.
2. **[tsml_regions_list]:** The link for a test region I have (tag_id=8] is [https://northmsaa.org/testsite/meetings/?tsml-day=any&tsml-region=8](https://northmsaa.org/testsite/meetings/?tsml-day=any&tsml-region=8) but should be [https://northmsaa.org/testsite/meetings/?region=fulton-ms](https://northmsaa.org/testsite/meetings/?region=fulton-ms). Again, looks like TSML UI expects a slug instead of the term ID.

**Comment by joshreisner** [comment]
mm good catch. i made TSML UI use slugs so the URLs would be more readable. 

i think we should try to fix this on the TSML side. if people agree we should create an issue

**Reply by anchovie91471** [reply]
Would it be better to change Legacy to use slugs too (I agree the URL is nicere that way) or bring TSML UI inline with legacy?

**Reply by joshreisner** [reply]
I would change these functions depending if TSML UI is selected

**Reply by anchovie91471** [reply]
Cool. Let me know when you create an issue and I’ll tackle it.
